Make script editor translatable

marcelbrueckner opened the pull request:
Introduces translatable keys for script editor analogous to automation editor's translation.

Resolves #3848.

marcelbrueckner committed:
Make script editor's script picker translatable (home-assistant/home-assistant-polymer#3848)

marcelbrueckner committed:
Make script editor translatable (home-assistant/home-assistant-polymer#3848)

homeassistant commented:
Hi @marcelbrueckner, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant-polymer#3866).

Once you do that we will be able to review and accept this pull request.

Thanks!

marcelbrueckner committed:
Fix linting errors (home-assistant/home-assistant-polymer#3866)

iantrich commented on src/panels/config/script/ha-script-editor.js:
```diff
@@ -254,7 +259,7 @@ class HaScriptEditor extends LocalizeMixin(NavigateMixin(PolymerElement)) {
       this.dirty &&
       // eslint-disable-next-line
-      !confirm("You have unsaved changes. Are you sure you want to leave?")
+      !confirm("[[localize('ui.panel.config.script.editor.unsaved_confirm')]]")
```
This one is likely more general than script editor

marcelbrueckner replied:
Should I put it below `ui.common` then? I have oriented myself on the automation editor's translation. In addition, basically every other section has it's own strings, too (just search for `delete_confirm`or `confirm_delete` in `en.json`).

iantrich replied:
Interesting...we should clean that up. We're just making more work for those who need to translate. I think maybe ui.panel.config.editor.common?

marcelbrueckner replied:
Alright. Will use `ui.panel.config.common.editor`, though, to keep up with the scheme `ui.panel.config.SECTION.editor`.

marcelbrueckner committed:
Fix linting errors (home-assistant/home-assistant-polymer#3866)

marcelbrueckner committed:
Move unsaved_confirm translation key to common section (home-assistant/home-assistant-polymer#3866)

Instead of adding the same text multiple times for every section, add a common section to indicate reusable translations.

bramkragten commented on src/panels/config/script/ha-script-picker.ts:
```diff
@@ -97,7 +113,9 @@ class HaScriptPicker extends LitElement {
-      message: `Triggered ${computeStateName(script)}`,
+      message: `${this.hass.localize(
+        "ui.dialogs.notification_toast.triggered"
+      )} ${computeStateName(script)}`,
```
Let's give `computeStateName(script)` as variable to lokalize, so the order can be changed by the translator:

```js
${this.hass.localize("ui.dialogs.notification_toast.triggered", "name", computeStateName(script))}
```

The string in `en.json`would be:
`Triggered {name}`

bramkragten commented on src/panels/config/script/ha-script-editor.js:
```diff
@@ -100,7 +100,10 @@ class HaScriptEditor extends LocalizeMixin(NavigateMixin(PolymerElement)) {
-            <div main-title>Script [[computeName(script)]]</div>
+            <div main-title>
+              [[localize('ui.panel.config.script.caption')]]
+              [[computeName(script)]]
```
Let's give `computeName(script)` as variable to lokalize, so the order can be changed by the translator:

```js
[[localize('ui.panel.config.script.caption'), 'name', computeName(script)]]
```

The string in `en.json`would be:
`Script {name}`

bramkragten commented on src/panels/config/script/ha-script-editor.js:
```diff
@@ -281,7 +286,9 @@ class HaScriptEditor extends LocalizeMixin(NavigateMixin(PolymerElement)) {
   async _delete() {
-    if (!confirm("Are you sure you want to delete this script?")) {
+    if (
+      !confirm("[[localize('ui.panel.config.script.editor.delete_confirm')]]")
```
Remove the "", it is no longer a string
```suggestion
      !confirm([[localize('ui.panel.config.script.editor.delete_confirm')]])
```

bramkragten replied:
And use `this.hass.localize`, as it is not used inside the Polymer render.

bramkragten commented on src/panels/config/script/ha-script-editor.js:
```diff
@@ -254,7 +259,7 @@ class HaScriptEditor extends LocalizeMixin(NavigateMixin(PolymerElement)) {
       this.dirty &&
       // eslint-disable-next-line
-      !confirm("You have unsaved changes. Are you sure you want to leave?")
+      !confirm("[[localize('ui.panel.config.common.editor.confirm_unsaved')]]")
```
Remove the ""

bramkragten commented on src/panels/config/script/ha-script-editor.js:
```diff
@@ -232,7 +235,9 @@ class HaScriptEditor extends LocalizeMixin(NavigateMixin(PolymerElement)) {
         () => {
-          alert("Only scripts inside scripts.yaml are editable.");
+          alert(
+            "[[localize('ui.panel.config.script.editor.load_error_not_editable')]]"
```
Remove the ""

marcelbrueckner replied:
If I remove the quotes, `npm run lint` throws following errors:

```
  238:17  error  Insert `[`                                                                                                                                                 prettier/prettier
  239:14  error  Replace `[localize('ui.panel.config.script.editor.load_error_not_editable')]]` with `localize("ui.panel.config.script.editor.load_error_not_editable")],`  prettier/prettier
  239:15  error  'localize' is not defined                                                                                                                                  no-undef
  240:11  error  Insert `]`                                                                                                                                                 prettier/prettier
```

After trying to fix the new linting issues, the error `'localize' is not defined` still remains.

bramkragten replied:
Yes, because it is not inside the render method of Polymer, use `this.hass.localize`

marcelbrueckner replied:
Thank you very much. I think I've got the point now.

bramkragten commented on src/panels/config/script/ha-script-editor.js:
```diff
@@ -244,7 +249,7 @@ class HaScriptEditor extends LocalizeMixin(NavigateMixin(PolymerElement)) {
     this.dirty = false;
     this.config = {
-      alias: "New Script",
+      alias: "[[localize('ui.panel.config.script.editor.default_name')]]",
```
Remove the ""

marcelbrueckner committed:
Add variable to localization text

marcelbrueckner committed:
Use JavaScript instead of Polymer data binding

bramkragten reviewed (approved):
Great thanks! 🦁
